azure-sink-connector/src/main/java/io/aiven/kafka/connect/azure/AzureBlobSinkConfig.java

```diff
@@ -0,0 +1,307 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.azure;
+
+import java.time.Duration;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
+
+import io.aiven.kafka.connect.common.config.AivenCommonConfig;
+import io.aiven.kafka.connect.common.config.CompressionType;
+import io.aiven.kafka.connect.common.config.FixedSetRecommender;
+import io.aiven.kafka.connect.common.config.OutputField;
+import io.aiven.kafka.connect.common.config.OutputFieldEncodingType;
+import io.aiven.kafka.connect.common.config.OutputFieldType;
+import io.aiven.kafka.connect.common.config.TimestampSource;
+import io.aiven.kafka.connect.common.config.validators.FilenameTemplateValidator;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class AzureBlobSinkConfig extends AivenCommonConfig {
+    private static final Logger LOG = LoggerFactory.getLogger(AzureBlobSinkConfig.class);
+    private static final String USER_AGENT_HEADER_FORMAT = "Azure Blob Sink/%s (GPN: Aiven;)";
+    public static final String USER_AGENT_HEADER_VALUE = String.format(USER_AGENT_HEADER_FORMAT, Version.VERSION);
+    private static final String GROUP_AZURE = "Azure";
+    public static final String AZURE_STORAGE_CONNECTION_STRING_CONFIG = "azure.storage.connection.string";
+    public static final String AZURE_STORAGE_CONTAINER_NAME_CONFIG = "azure.storage.container.name";
+    public static final String AZURE_USER_AGENT = "azure.user.agent";
+    private static final String GROUP_FILE = "File";
+    public static final String FILE_NAME_PREFIX_CONFIG = "file.name.prefix";
+    public static final String FILE_NAME_TEMPLATE_CONFIG = "file.name.template";
+    public static final String FILE_COMPRESSION_TYPE_CONFIG = "file.compression.type";
+    public static final String FILE_MAX_RECORDS = "file.max.records";
+    public static final String FILE_NAME_TIMESTAMP_TIMEZONE = "file.name.timestamp.timezone";
+    public static final String FILE_NAME_TIMESTAMP_SOURCE = "file.name.timestamp.source";
+
+    public static final String FORMAT_OUTPUT_FIELDS_CONFIG = "format.output.fields";
+    public static final String FORMAT_OUTPUT_FIELDS_VALUE_ENCODING_CONFIG = "format.output.fields.value.encoding";
```

Comment:
AFAIU this doesn't have a define() call. Maybe some other configurations are missing them too?
If define() isn't called, the configuration is effectively hidden from all tooling.

Reply:
The define is in parent class. Not sure why we need to copy-paste this static field however